rscripts/run_shiny_server.R

```diff
@@ -0,0 +1,755 @@
+                ),
+
+                tabPanel(
+                  "3nt periodicity",
+                  headerPanel(""),
+                  plotOutput("periodicity_line_plot"),
+                  headerPanel(""),
+                  plotOutput("frame_counts_bar_plot"),
+                  headerPanel(""),
+                  plotOutput("frame_proportions_boxplot")
+                ),
+
+                tabPanel(
+                  "Normalized reads by position",
+                  headerPanel(""),
+                  plotOutput("pos_sp_plot5"),
+                  headerPanel(""),
+                  plotOutput("pos_sp_plot3")
+                ) ,
+
+                tabPanel(
+                  "Gene specific coverage",
+                  wellPanel(textInput(
+                    inputId = "gene",
+                    label = "Gene:",
+                    value = collated_tpms_df$ORF[1],
+                    width = "50%"
+                  )),
+                  uiOutput("gene_specific_slider"),
+                  headerPanel(""),
+                  plotOutput("sgribogrid_plot")
+                ),
+
+                tabPanel(
+                  "Ribogrid",
+                  sliderInput(
+                    "ribogrid_len_range",
+                    "Read lengths:",
+                    min = 0,
+                    max = 50,
+                    value = c(20, 35),
+                    width = "50%"
+                  ),
+                  sliderInput(
+                    "ribogrid_pos_range",
+                    "Read position:",
+                    min = -24,
+                    max = 50,
+                    value = c(-10, 10),
+                    width = "50%"
+                  ),
+                  headerPanel(""),
+                  plotOutput("ribogrid_plot"),
+                  headerPanel(""),
+                  plotOutput("ribogridbar_plot", height = "1000px")
+                ),
+
+                tabPanel(
+                  "Features",
+                  wellPanel(textInput(
+                    inputId = "gene2",
+                    label = "Gene:",
+                    value = collated_tpms_df$ORF[1],
+                    width = "50%"
+                  ),
+                  if (any(names(yaml) == "features_file") &&
+                      length(yaml$features_file) > 0) {
+                    checkboxGroupInput(
+                      inputId = "feature",
+                      label = "Feature:",
+                      choiceNames = possible_features,
+                      choiceValues = possible_features,
+                      selected = possible_features[1],
+                      inline = TRUE,
+                      width = "100%"
+                    )
+                  }),
+                  headerPanel(""),
+                  plotOutput("features_plot", height = "1000px")
+                )
+              )
+            )))
+
+# I cast spell, ~run shiny app~
+shinyApp(ui, server, options = list(port = port_num))
```

Comment:
Is it a problem if there is no new line at the end of a file? It seems to be flagged with a little stop sign so I thought I'd call attention to it.